### DISCUSSION TITLE
react-native-template-new-architecture

### DISCUSSION THREAD

**Discussion by leotm** [reactwg/react-native-new-architecture#21] [Libraries]
[leotm/react-native-template-new-architecture](https://github.com/leotm/react-native-template-new-architecture)
_Originally forked from [react-native-community](https://github.com/react-native-community/react-native-template-typescript) ❤️_

### RN 0.69.0
- [x] Hermes
- [x] Fabric
- [x] TurboModules
- [x] React 18

### React Native
- [x] <s>0.68.0-rc.4</s>
- [x] <s>0.68.0 stable</s>
- [x] <s>0.68.1 stable</s>
- [x] <s>0.69.0-rc0</s>
- [x] <s>0.69.0-rc1</s>
- [x] <s>0.69.0-rc6</s>
- [x] 0.69.0 stable
- [ ] 0.69.1

### Latest/future (unsupported) architecture
- [x] M1 (arm64)
- [x] <s>NDK 24 rc3</s>
- [x] <s>NDK 25-rc2</s>
- [x] <s>NDK 25-rc3</s>
- [x] NDK 25-rc4 🔥
- [x] SDK 33-rc2 🔥
- [ ] SDK 33 stable
- [x] JDK (aarch64)
  - [x] openjdk-18 🔥
  - [x] zulu-18
  - [x] sapmachine-18
  - [x] liberica-18 🐌
  - [x] corretto-17
  - [x] temurin-17

_Official: [environment-setup](https://reactnative.dev/docs/environment-setup)_
- https://github.com/facebook/react-native/issues/34103

### With few bleeding edge-tras
- [x] <s>TypeScript 4.6 nightly</s>
- [x] <s>TypeScript 4.7 Beta</s>
- [x] <s>TypeScript 4.7 stable</s>
- [x] TypeScript 4.8 Beta 🔥
- [x] ESLint 8 (_opinionated/all/unstable [rules](https://github.com/leotm/react-native-template-new-architecture/blob/master/.eslintrc.js)_)
- [x] Prettier 2
- [x] Renovate
- [x] RN Storybook v5 w server
- [x] RN Storybook v6 alpha
- [ ] RN Storybook v6 [beta](https://github.com/storybookjs/react-native/releases?q=beta&expanded=true)
- [x] Yarn [3.2](https://dev.to/arcanis/yarn-32-libc-yarn-explain-next-major--o22)
- [ ] Yarn 4
- [x] RN Flipper _latest_ w node server
- [x] TS Node / Deno examples
- [x] GitHub Actions workflow
  - [x] Static analysis
  - [x] Android on Ubuntu
  - [ ] iOS on macOS-12
- [x] Wiki
- [x] Project
- [x] Security
- [x] Labels

### Xcode
- [x] <s>Xcode 13.3</s>
- [x] <s>Xcode 13.3.1</s>
- [x] <s>Xcode 13.4</s>
- [x] Xcode 14 beta 1 🔥

### Ubuntu
- [x] Ubuntu 20.04 🚀
- [ ] Ubuntu 22.04 beta

### macOS 12
- [x] <s>Monterey 12.3</s>
- [x] <s>Monterey 12.4 Beta</s>
- [x] <s>Monterey 12.4 Beta 2</s>
- [x] <s>Monterey 12.4 Beta 3</s>
- [x] <s>Monterey 12.4 Beta 4</s>
- [x] <s>Monterey 12.5 Beta</s>
- [x] <s>Monterey 12.5 Beta 2</s>
- [x] Monterey 12.5 Beta 3 🔥

### macOS 13
- [ ] Ventura 13.0 Beta 2

### Gradual adoption
- [x] `react-navigation`, `react-native-screens` and `react-native-safe-area-context`
  - [x] M1 workaround https://github.com/reactwg/react-native-releases/discussions/13#discussioncomment-2269318
  - [x] <s>[Gradle Wrapper clean issue](https://github.com/reactwg/react-native-releases/discussions/13#discussioncomment-2269223)</s>
  - [ ] _Demo app_ branch
- [ ] `react-native-gesture-handler`
- [ ] `react-native-webview`
- [ ] `react-native-reanimated`
- [ ] `react-native-pager-view`
- [ ] `react-native-gesture-handler`
- [ ] `rnx-kit`
- [ ] ` mobile-dev-inc/dadb`

### Issues
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/482
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/521
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/525
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/526
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/553
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/587
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/598
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/602
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/613
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/618
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/636
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/637
- [x] https://github.com/leotm/react-native-template-new-architecture/issues/639
- [ ] TODO: Update

<s>Will adjust docs for speedier Android builds when merged</s>
- [x] https://github.com/facebook/react-native-website/pull/3027

**Comment by cortinico** [comment]
Hi @leotm 
I suppose this is a status update on the migration of `react-native-template-typescript` to the New Architecutre right? Could you edit your message and provide a better description on top as it's not immediately clear from reading this.

Also is the progress available on a branch on Github that users can take a look at?

**Reply by leotm** [reply]
Sure adjusted the title ^ the migration's enabled/available on `master`, including react-navigation and react-native-screens

But still in need of ReadMe tweaks i'm realising from following discussions (learning alot 🙌) and integrating libraries

**Reply by leotm** [reply]
That's a good idea, i'll tidy then link my branch with my navigation demo app ([FabricExample](https://github.com/software-mansion/react-native-screens/tree/main/FabricExample) integrated)

And create/link branches for minimal demo apps testing integration of the above libraries and continue collabing on an issues spotted while following maintainer updates

---

Didn't realise you're in London! I'm running 5K every couple days around Wembley too, gone from barefoot to vibrams

**Reply by cortinico** [reply]
> the migration's enabled/available on `master`

Which `master`? I see https://github.com/react-native-community/react-native-template-typescript doesn't have a `master` branch. Are you talking about a different repo?

**Reply by leotm** [reply]
Yes different repo: https://github.com/leotm/react-native-template-typescript

Updated description above

And thanks for updating title ^

**Reply by cortinico** [reply]
Thanks for the update. Before you fork [react-native-community/react-native-template-typescript](https://github.com/react-native-community/react-native-template-typescript), have you considered contributing to the original repo?

I would start from opening a issue there and ask if you can contribute to their repo.

I appreciate your effort in creating an alternative, but I don't think we want to point users to a forked solution unless there is a valid strong opinion to pick your fork instead of `react-native-community/react-native-template-typescript`.

**Reply by leotm** [reply]
Np, yh I contributed couple yrs ago, albeit only a commit

I forked cos I don't think my wild opinionated ideas in combination constitute

> Clean and minimalist React Native template for a quick start with TypeScript

**Reply by leotm** [reply]
Shall I continue to open an issue for this to contribute more?

(Last time I didn't open an issue, but just PR'ed which @emin93 accepted if i remember right)

I just don't want to force my opinions on the community pkg, but rather offer more (I'm looking at building a CLI)

I'm not trying to point anybody away from the OG that inspired me and the rest of us TypeScript fanbois/girls 😅 (sry Flow)

edit: [also](https://github.com/react-native-community/react-native-template-typescript/pull/255#issuecomment-1070963965)

> we link this [react-native-community] template to official RN releases and not to the main branch. Otherwise the template could be incompatible with the stable release

my `master`'s ahead on 0.68.0-rc3 (i.e. bleeding-edge edition/version)

**Reply by cortinico** [reply]
Hey @leotm,
I appreciate your effort in building this template. However, I suggest you to rename your repo to something other than `react-native-template-typescript`. 

The reason is that your template is doing a bit more than just adding TypeScript support and I believe this could be confusing for your users and for the community in generally.

How do you feel about it?

**Reply by leotm** [reply]
Sure, feelin' good ^ done

**Comment by radko93** [comment]
Thanks for your work @leotm.
We have updated the official template with RN 0.68.0-rc.4 and everyone can try that by using 6.10.0-rc.1 or use `next` tag (versions are not tied to RN versions as we tend to have more updates for the template). We will update the template when the stable release is out. 

To try the latest pre-release version:
```sh
npx react-native init MyApp --template react-native-template-typescript@next
```

We try to keep the template as close as possible to the core template by only adding a small subset of packages required for Typescript to work. By doing this, we can ensure that developer experience is the same as when using bare RN init command.

**Reply by leotm** [reply]
Sweet ^ saw last night but fell asleep half way thru [reviewing](https://github.com/react-native-community/react-native-template-typescript/pull/258/files) sry, great stuff ☝️

**Reply by leotm** [reply]
My plans are more providing [Re.Pack](https://github.com/callstack/repack) w [Module Federation](https://github.com/callstack/repack/projects/3), ClojureScript, Speedy Web Compiler, a [FastLane BUCK action/plugin](https://github.com/fastlane/fastlane/issues/7990#issuecomment-282843798) that was never done, [rnx-kit](https://github.com/microsoft/rnx-kit), React Concurrent and Suspense, etc and ripping the template apart more while upgrading everything to unstable versions for early adopters to spot/fix issues way ahead (_not production_) - then eventually run [DOOM on it with WebAssembly](http://35.158.218.205/experiments/webDOOM/) as part of the unofficial [FB-MSFT Collaboration](https://github.com/react-native-community/discussions-and-proposals/projects/1) [:suspect:](https://youtu.be/KN0K58EfJSg) and provide a CLI for stable/production

Keep up the good work

**Reply by radko93** [reply]
That's sounds exciting, I like the native additions because most of the JS side changed to the template are usually very opinionated (I don't use things from your template in my projects).
